test/backend/non_zero.in.cpp

```diff
@@ -0,0 +1,105 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/all_close_f.hpp"
+#include "util/ndarray.hpp"
+#include "util/test_control.hpp"
+#include "util/test_tools.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+
+NGRAPH_TEST(${BACKEND_NAME}, non_zero)
+{
+    PartialShape p_shape = PartialShape::dynamic();
+    auto p = make_shared<op::Parameter>(element::f32, p_shape);
+    auto non_zero = make_shared<op::v3::NonZero>(p, element::i32);
+    auto fun = make_shared<Function>(OutputVector{non_zero}, ParameterVector{p});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+    auto cfun = backend->compile(fun);
+
+    auto input = backend->create_tensor(element::f32, Shape{3, 2});
+    copy_data(input, vector<float>{0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 3.0f});
+
+    std::vector<int32_t> expected_result{2, 2, 0, 1};
+    Shape expected_output_shape{2, 2};
+
+    auto result = make_shared<HostTensor>();
```

Comment:
This won't work for non-CPU devices since you didn't use the backend to create the tensor. @rkimballn1 what is the best way to deal with this?

Reply:
Scott is correct, the CPU backend will cast this to a CPU Tensor which it is not.
We need to call backend->create_tensor() with no arguments. The backend needs to support this which none of the backends currently support. 
For now put this test in the CPU manifest and we can do a separate PR to add the needed backend API for creating tensors of unknown size/type